libs/dyn/value.go

```diff
@@ -2,13 +2,29 @@ package dyn
 
 import (
 	"fmt"
+	"slices"
 )
 
 type Value struct {
 	v any
 
 	k Kind
-	l Location
+	// TODO: Format ugly pieces of patched code.
+
+	// Locations is a list of locations that this value was defined at or derived from.
+	// This metadata is maintained across merging of configuration trees, which, for example
+	// could be due to includes or target overrides in the YAML configuration.
+	//
+	// The last location in the list is the "effective" location for the value.
+	// For scalar values this will correspond to the location the value was defined at
+	// in the original YAML configuration (if any).
+	// For maps and sequences, merge is additive and can override existing values.
+	// so there are no particular semantics associated with the effective location.
```

Comment:
What can be expected? I.e. the first location they showed up in?

Reply:
Yeah. The first location they show up in. Redid the code a bit, this aspect should be clearer now.